internal/incident/incident.go

```diff
@@ -73,51 +75,70 @@ func (i *Incident) HasManager() bool {
 }
 
 // ProcessEvent processes the given event for the current incident.
-func (i *Incident) ProcessEvent(ev event.Event, created bool) error {
+func (i *Incident) ProcessEvent(ctx context.Context, tx *sqlx.Tx, ev event.Event, created bool) error {
 	i.Lock()
 	defer i.Unlock()
 
 	i.runtimeConfig.RLock()
 	defer i.runtimeConfig.RUnlock()
 
+	err := i.Object.UpdateMetadata(ctx, tx, ev.SourceId, ev.Name, utils.ToDBString(ev.URL), ev.ExtraTags)
+	if err != nil {
+		i.logger.Errorw("Can't update object metadata", zap.Error(err))
+
+		return errors.New("can't update object metadata")
+	}
+
+	if ev.ID == 0 {
+		eventRow := event.NewEventRow(ev, i.Object.ID)
+		eventID, err := utils.InsertAndFetchId(ctx, tx, utils.BuildInsertStmtWithout(i.db, eventRow, "id"), eventRow)
+		if err != nil {
+			i.logger.Errorw("Failed to insert event and fetch its ID", zap.Error(err))
+
+			return errors.New("can't insert event and fetch its ID")
+		}
+
+		ev.ID = eventID
+	}
```

Comment:
Is there any benefit in doing this here compared to where it was done before (i.e. in listener)? At least `UpdateMetadata()` hasn't much to do with the current incident, so I don't think this makes much sense in a receiver function on the incident type.

Similarly for writing the event to the database. At the moment it's tied to an incident, but in the future, also events like downtime starts will be handled, and these can occur without an active incident and should also be written to the database, so I wouldn't move this here either.